extensions/traceability.sigmf-ext.md

```diff
@@ -0,0 +1,201 @@
+# The `traceability` SigMF Extension Namespace v1.0.0
+
+This document proposes a new extension namespace called `traceability` for the Signal Metadata
+Format (SigMF) specification. This extension provides traceability information for the metadata.
+
+## 1 Global
+
+`traceability` extends the [Global](https://github.com/gnuradio/SigMF/blob/master/sigmf-spec.md#global-object) object.
+
+The following fields are added to the `global` object:
+
+|name|required|type|description|
+|----|--------|----|-----------|
+|`traceability:last_modified`|false|[DataChange](#datachange-object)|Captures the author and timestamp of the most recent modification|
+|`traceability:last_reviewed`|false|[DataChange](#datachange-object)|Captures the author and timestamp of the most recent review|
+|`traceability:revision`|false|integer|Specifies the revision number associated with the metadata|
+|`traceability:origin`|false|[Origin](#origin-object)|Provides information about the origin of the data|
+|`traceability:sample_length`|false|integer|Specifies the total number of samples of the Dataset|
+
+### DataChange Object
+
+|name|required|type|description|
+|----|--------|----|-----------|
+|`author`|false|string|Email address of the author who changed the metadata|
+|`time`|true|string (date-time)|Timestamp of the modification in ISO 8601 format|
+
+### Origin Object
+
+|name|required|type|description|
+|----|--------|----|-----------|
+|`type`|true|string|Type of origin information. Ex.: `blob`,`local`,`memmory`|
```

Comment:
@Nepomuceno can you elaborate on the purpose of this object? Its marked as "required" yet does not appear in the examples below.

Reply:
What does `local` and `memory` even mean, it's a relative thing.  Now if let's say the file was stored on a NAS instead of cloud storage, I can see someone wanting to put in their hostname or some identifier of the NAS or file server, instead of an account and container name.

Reply:
Ok, so this is basically a free-form field for users to add some extra data - im fine keeping this but lets make it optional.

Reply:
Well what I'm getting at is that we don't need the field, it doesn't convey enough meaningful info imo

Reply:
Yeah, I wouldn't really find it useful either though if there's a specific need I'm fine with keeping it as optional. Also definitely ok with removing it.